doc/reports/spikes/metrics-what-we-want-spike.md

```diff
@@ -0,0 +1,18 @@
+# Monitoring Performance - What should we monitor?
+## Introduction
+This spike report follows on from work and discussions in the OAT meeting to identify areas where we want to be monitoring the system. This outlines what I see as an idealised monitoring setup - there may be blockers to overcome to achieve some of this. I've also provided guidance in the document to how I see the monitoring being provided and used.
+## Production metrics
+All of these metrics should be radiated on monitors
+### Performance
+* **99% call time** - This can be pulled from Google Analytics, CloudWatch or Zipkin headers in Splunk logs, potentially further broken down by microservice
+* **Current concurrent users** - This can be pulled most easily from Google Analytics
+* **Peak concurrent users** - As above, but the maximum for the last 24h/72h/7d
+### Physical
+* **Diego cell usage** - This can be pulled from Cloudwatch. It's a candidate for alerting into Slack (and does currently alert), but note that Diego Cell balancing can often be poor (consider alerting if a significant number of DCs are maxed or the average load exceeds x%)
+* **Database CPU%** - This can be pulled from Cloudwatch or by passing DB physical metrics into Splunk. This is a very good candidate for alerting into Slack, as going above 80% is usually a precursor to a collapse in system performance
+### Platform 
+* **App crashes** - This can be pulled from CloudFoundry or Splunk. It can be a candidate for alerting, but would need to be set to an appropriate threshold - single app crashes are usually fine and don't require investigation while multiple in short order need flagging up
```

Comment:
Does this include app downtime? If so I think we need to be clearer here. We will also want to monitor if the internal site (ROps) is not available too as we currently alert when frontstage is down.

Reply:
Further to Rhi's point, we need to check that healthcheck endpoints are are reflective of health, as I think recently one was reporting healthy when it wasn't.

Reply:
We could measure app downtime if we felt it useful but it's not what I'm raising here. I'm suggesting that any app that crashes more than a couple of times in an hour should be alerted and investigated.